mlflow/models/evaluation/base.py

```diff
@@ -1321,11 +1321,22 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
 
     _EnvManager.validate(env_manager)
 
-    if model_type in [_ModelType.REGRESSOR, _ModelType.CLASSIFIER] and targets is None:
-        raise MlflowException(
-            f"The targets argument must be specified for {model_type} models.",
-            error_code=INVALID_PARAMETER_VALUE,
-        )
+    if model_type in [_ModelType.REGRESSOR, _ModelType.CLASSIFIER]:
+        if isinstance(data, Dataset):
+            if hasattr(data, "targets") and data.targets is not None:
```

Comment:
```suggestion
            if (targets := getattr(data, "targets")) is not None:
```

We can use getattr and assignment expression here.